### PR TITLE
Fixes a bug in TorsionConstraint

### PR DESCRIPTION
**Pull request by ptosco**
This bug was reported by Thomas Fox:
https://www.mail-archive.com/rdkit-discuss@lists.sourceforge.net/msg09149.html

The bug is not in `setDihedralDeg()`, but in the dihedral constraint code.
While reviewing that code I found some code duplication that I decided to remove.
Also, I noticed discrepancies between the C++ and Python unit tests for constraints, and I made them consistent.
I fixed the dihedral ranges in some tests where they were incorrect.

### CODE REVIEW

File: Code/ForceField/ForceField.cpp [hunk 1]
```diff
@@ -14,6 +14,80 @@
...
+  RDGeom::Point3D p4(pos[3 * idx4], pos[3 * idx4 + 1],
+                     pos[3 * idx4 + 2]);
+  computeDihedral(&p1, &p2, &p3, &p4, dihedral, cosPhi, r, t, d);
+}
+
+void computeDihedral(const RDGeom::Point3D *p1, const RDGeom::Point3D *p2,
+    const RDGeom::Point3D *p3, const RDGeom::Point3D *p4,
+    double *dihedral, double *cosPhi, RDGeom::Point3D r[4],
+    RDGeom::Point3D t[2], double d[2]) {
+  RDGeom::Point3D rLocal[4];
```
Comment (greglandrum) [inline Code/ForceField/ForceField.cpp]: I think you need PRECONDITIONS on p1, p2, p3, and p4 making sure they aren't `nullptr`

File: Code/ForceField/MMFF/TorsionConstraint.h [hunk 2]
```diff
@@ -55,6 +55,10 @@ class RDKIT_FORCEFIELD_EXPORT RDKIT_FORCEFIELD_EXPORT TorsionConstraintContrib
   };
 
  private:
+  inline void setParameters(ForceField *owner, unsigned int idx1,
```
Comment (greglandrum) [inline Code/ForceField/MMFF/TorsionConstraint.h]: why `inline`?

Comment (ptosco) [reply Code/ForceField/MMFF/TorsionConstraint.h]: I added the `setParameters` function to remove some code duplication, but as it is literally only assigning parameters which should not result in a large chunk of code I thought inlining would be good (the compiler will decide for the best anyway). Happy to remove it if you prefer.

File: Code/ForceField/ForceField.cpp (hunk 1 above)
Comment (ptosco) [reply Code/ForceField/ForceField.cpp]: Good point, I will add them.

File: Code/ForceField/MMFF/TorsionConstraint.h (hunk 2 above)
Comment (greglandrum) [reply Code/ForceField/MMFF/TorsionConstraint.h]: just doesn't seem like a particularly useful recommendation to the compiler in this case
but I don't feel strongly about it

**Comment by ptosco** [comment]
@greglandrum Done.

**Pull request merged**